src/tests/dbus-tests/test_zram.py

```diff
@@ -0,0 +1,181 @@
+import os
+import re
+import time
+
+import storagedtestcase
+
+
+MODPROBECONF = '/usr/lib/modprobe.d/zram.conf'
+MODLOADCONF = '/usr/lib/modules-load.d/zram.conf'
+ZRAMCONFDIR = '/usr/local/lib/zram.conf.d'
+
+
+class StoragedZRAMTest(storagedtestcase.StoragedTestCase):
+    '''This is a basic ZRAM test suite'''
+
+    conf = {MODPROBECONF: None,
+            MODLOADCONF: None,
+            ZRAMCONFDIR: {}}
+
+    @classmethod
+    def _save_conf_files(cls):
+        # /usr/lib/modprobe.d/zram.conf
+        if os.path.exists(MODPROBECONF):
+            cls.conf[MODPROBECONF] = cls.read_file(MODPROBECONF)
+
+        # /usr/lib/modules-load.d/zram.conf
+        if os.path.exists(MODLOADCONF):
+            cls.conf[MODLOADCONF] = cls.read_file(MODLOADCONF)
+
+        # /usr/local/lib/zram.conf.d
+        if os.path.exists(ZRAMCONFDIR):
+            for fname in os.listdir(ZRAMCONFDIR):
+                cls.conf[ZRAMCONFDIR][fname] = cls.read_file(ZRAMCONFDIR + '/' + fname)
+
+    @classmethod
+    def _restore_conf_files(cls):
+        # /usr/lib/modprobe.d/zram.conf
+        if cls.conf[MODPROBECONF]:
+            cls.write_file(MODPROBECONF, cls.conf[MODPROBECONF])
+
+        # /usr/lib/modules-load.d/zram.conf
+        if cls.conf[MODLOADCONF]:
+            cls.write_file(MODLOADCONF, cls.conf[MODLOADCONF])
+
+        # /usr/local/lib/zram.conf.d
+        for fname in cls.conf[ZRAMCONFDIR].keys():
+            cls.write_file(ZRAMCONFDIR + '/' + fname, cls.conf[ZRAMCONFDIR][fname])
+
+    @classmethod
+    def setUpClass(cls):
+        storagedtestcase.StoragedTestCase.setUpClass()
+        cls.ensure_modules_loaded()
+
+        cls._save_conf_files()
+
+    @classmethod
+    def tearDownClass(cls):
+        storagedtestcase.StoragedTestCase.tearDownClass()
+
+        cls._restore_conf_files()
+
+        cls.run_command('rmmod zram')
+
+    def _get_zrams(self):
+        time.sleep(0.5)
+        udisks = self.get_object('')
+        objects = udisks.GetManagedObjects(dbus_interface='org.freedesktop.DBus.ObjectManager')
+
+        return [path for path in objects.keys() if re.match(r'.*/block_devices/zram[0-9]$', path)]
```

Comment:
It might be slightly better to use "[0-9]+$" in regexp in case there are more than 9 zrams present.